app/assets/javascripts/short-form/ShortFormApplicationService.js.coffee.erb

```diff
@@ -505,9 +505,14 @@ ShortFormApplicationService = (
   Service.signInSubmitApplication = (opts = {}) ->
     # check if this user has already applied to this listing
     Service.getMyAccountApplication().success((data) ->
-      if !_.isEmpty(data.application) && Service._previousIsSubmittedOrBothDrafts(data.application)
+      previousApplication = data.application
+      if !_.isEmpty(previousApplication)
         # if user already had an application for this listing
-        return Service._signInAndSkipSubmit(data.application)
+        if opts.type == 'review-sign-in' && previousApplication.status.match(/draft/i)
+          # override draft ID and proceed, so that we eventually call submitApplication...
+          Service.overridePreviousDraftId()
```

Comment:
Is overriding the draft ID with that was already saved happen after then choose which draft to use?